src/Microsoft.TestPlatform.CommunicationUtilities/SocketCommunicationManager.cs

```diff
@@ -153,20 +161,39 @@ public void StopServer()
         /// <summary>
         /// Connects to server async
         /// </summary>
-        /// <param name="portNumber">Port number for client to connect</param>
+        /// <param name="endPointAddress">EndPointAddress for client to connect</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
-        public async Task SetupClientAsync(int portNumber)
+        public async Task SetupClientAsync(string endPointAddress)
         {
             this.clientConnectionAcceptedEvent.Reset();
-            EqtTrace.Info("Trying to connect to server on port : {0}", portNumber);
+            var endPoint = endPointAddress.Split(':');
+            EqtTrace.Info("Trying to connect to server on socket : {0} : {1}", endPoint[0], endPoint[1]);
             this.tcpClient = new TcpClient();
             this.socket = this.tcpClient.Client;
-            await this.tcpClient.ConnectAsync(IPAddress.Loopback, portNumber);
-            this.stream = this.tcpClient.GetStream();
-            this.binaryReader = new BinaryReader(this.stream);
-            this.binaryWriter = new BinaryWriter(this.stream);
-            this.clientConnectionAcceptedEvent.Set();
-            EqtTrace.Info("Connected to the server successfully ");
+
+            Stopwatch watch = new Stopwatch();
+            watch.Start();
+            do
+            {
+                try
+                {
+                    await this.tcpClient.ConnectAsync(IPAddress.Parse(endPoint[0]), int.Parse(endPoint[1]));
+
+                    if (this.tcpClient.Connected)
+                    {
+                        this.stream = this.tcpClient.GetStream();
+                        this.binaryReader = new BinaryReader(this.stream);
+                        this.binaryWriter = new BinaryWriter(this.stream);
+                        EqtTrace.Info("Connected to the server successfully ");
+                        this.clientConnectionAcceptedEvent.Set();
+                    }
+                }
+                catch (Exception ex)
```

Comment:
I'll mark a ToDo here to pass cancellation token, in case user cancel's the connection operation. So we don't spend additional time to connect